### ISSUE TITLE
Ref DB index management in not up to current bioinformatics standards and is not adapted to large scale analysis in production workflows on HPC systems

### ISSUE THREAD

**Issue by ppericard** [biocore/sortmerna#233]
Right now, indexing the ref DB is done automatically by SMR but in a way that is not up to current bioinformatics standards and is not adapted to large scale analysis in production workflows on HPC systems.
The main concern is that it is not possible to easily give SMR a pre-computed index. Additional limitations are not using the standard basename scheme and writing in the home directory by default.
This new way of automatically dealing with indexing in SMR v3+ started with the best intentions but ended up being a regression.

Having an index allows for a decrease of the seed search complexity and therefore a speed-up of that search. However the indexing part can initially take time, and this is ok because it's a trade-off where we know indexing only has to be done once compared to the thousands of datasets that need to be analyzed.

To do that, modern aligners like STAR, Minimap,  Bowtie2, etc, have 2 modes. The first mode to build the index from a Fasta file. The index contains all the files needed for the alignment part (sequences included) and is usually designed by a file basename set by the user (ex. Silva_128_rRNA) which will be used to name the index files (Silva_128_rRNA.index_1, Silva_128_rRNA.index_2, ...).
For a fixed alignment application/analysis, indexing needs to be done only once. The users will choose/build the ref db they want to work with, build the index once and for all, and then start aligning possibly hundred or thousands samples with it. This is especially true for SMR where indexing can take a long time on big reference databases. The index will seldom need to be updated compared to the number of times it will be used.
In most bioinformatics HPC systems, these indexes are even pre-computed by the system, stored in a specific storage system and made available to all users (see https://biomaj.genouest.org/ for example).

In SMR v2.1, this was dealt with by having 2 binaries (`indexbd_rna` for indexing and `sortmerna` for filtering/aligning). The main pb from a user point of view at that point was the way both the fasta file AND the index basename needed to be passed to `sortmerna` in a single parameter (`--ref ref.fa;index_basename`). This lead to many pbs linked to parameter parsing and bad handling of relative paths.
One way to improve SMR at that point would have been to store the sequences from the Fasta file in the index, in a compressed format (maybe gzipped) and possibly indexed itself (like a *.fai file from samtools index). Therefore, the user would only need to pass to "sortmerna" the basename of the index and everything would be available.
Like: 
```
sortmerna --ref /path/to/index/index_basename
sortmerna --ref index_basename(default to cwd)
sortmerna --ref ../../index_basename
```

where the index files would be something like:
```
index_basename.bursttrie_0
index_basename.kmer_0
index_basename.pos_0
index_basename.stats
index_basename.sequences (gzipped fasta)
```

It is still possible to do something like this in the new SMR version, with only one binary ("sortmerna"). The modern way to do it would be using subcommands. Something like:

```
sortmerna index --ref ref.fasta --index index_basename
sortmerna align --reads reads.fq --index index_basename
```

In any case, SMR should be able to support these functionnalities in the future:
- Easy indexing and naming of the index files. Tailored for a single use.
- Users should be able to easily pass to SMR a pre-computed index, refered by its basename in a relative path
- It would be better if the index was also storing the sequences. Users wouldn't need to pass the index AND the fasta file (which can lead to even more difficulties when the index does not correspond to the fasta file, or if they are not stored at the same location)

I'm addressing the defaut location of the index files (in the Home directory) in another issue.

**Comment by ppericard** [comment]
BTW using subcommands would also allow for interesting use cases of SMR if we added a "filter" subcommand, since this is one of the most used function of SMR:
```
sortmerna filter --reads reads.fq --index basename
```

This would default to outputing only the fastx output files and setting parameters to optimize speed (like --num_align 1, output reads as soon as one alignment passing the e-value threshold is found).

You would then have:
```
sortmerna index --ref ref.fasta --index index_basename
sortmerna filter --reads reads.fq --index basename (optimized for filtering, reduced set of parameters)
sortmerna align --reads reads.fq --index index_basename (full functionnality)
```
